### ISSUE TITLE
Wrong Julia used by SpineOpt after julia's update

### ISSUE THREAD

**Issue by tbaptistal** [spine-tools/Spine-Toolbox#2994]
I downloaded SpineToolBox and downloaded Julia (1.10) and SpineOpt from the Spinetoolbox settings < tool interface. 

But I needed to access the latest version of SpineOpt (which was not published at the time, and so I had to directly clone the repository). To stay coherent with the same installation tutorial, I removed Julia from my laptop and downloaded it again with python. I cloned SpineOpt repository and followed the installation procedure. 
Then I linked my new Julia (1.11) and the environment to get the latest version of SpineOpt: 
![image](https://github.com/user-attachments/assets/fdc962aa-4ac7-44b4-9ab5-f6c64b854706)

New project would generate the SpineOpt blocks following the right path to Julia 1.11 and the new SpineOpt in the Julia environment.
![image](https://github.com/user-attachments/assets/afa5afc8-2a2a-4dea-9919-39ac7407a1f5)

But old project like as follow, would continue to use julia 1.10.
![image](https://github.com/user-attachments/assets/8997a4ac-0091-477b-af17-809e0d46838c)

Even new blocks generated would follow the julia 1.10: 
![image](https://github.com/user-attachments/assets/c2607e5e-73ca-494b-8650-2c3621ce3e3a)

My general settings in this old project are not changed as follow:
![image](https://github.com/user-attachments/assets/43aa7edf-685a-4acc-9cf1-715c4562b9aa)

To remove this bug from the block in this old project, I had to manually change the specification of SpineOpt blocks by changing the _executable_ and the _project_ in the block:
![image](https://github.com/user-attachments/assets/adabb680-52a7-4b2d-b36a-62e825dd318d)

**Comment by soininen** [comment]
Right. Transferred the issue to Spine-Toolbox.

**Comment by ptsavol** [comment]
TLDR: This is probably fixed but please reopen if you find any other problems related to this.

I think I know what happened here. The directory `C:\<user_dir>\AppData\Local\julias\bin` contains symlinks to Julias that you have installed using Jill (Spine Toolbox's **Settings->Tools->Install Julia** button uses Jill too). When You (@tbaptistal) removed 
Julia 1.10, I assume that the symlink for that Julia was not removed. Then you installed Julia 1.11 using the Julia Install Wizard and then clicked Ok with the `Use this Julia with Spine Toolbox` checkbox selected and here was the bug. The symlink path that the wizard returned was the path to julia-1.10.cmd when it should have been julia-1.11.cmd. I fixed this so that it now always returns the path to `julia.cmd` on Win or `julia` on Linux and others. As the jill documentation says, `julia.cmd` or `julia` symlink in the bin folder always points to the latest Julia release (that you have installed). However, you can use `jill switch julia-x.x` to change the Julia that `julia.cmd` or `julia` launches.
